src/ca/noxid/lab/script/TscPane.java

```diff
@@ -309,12 +309,11 @@ private void setCommandExtras(int commandIndex, String cmd) {
 		if (selCom.numParam <= 0) {
 			return;
 		}
-
-		int commandSize = 4 + selCom.numParam * 5 - 1;
+		int commandSize = selCom.paramLen + selCom.numParam * (selCom.paramSep ? selCom.paramLen + 1 : selCom.paramLen) - 1;
 		if (selCom.numParam < 2) commandSize -= 1;
 		if (cmd.length() > commandSize) {
 			for (int i = 0; i < selCom.numParam; i++) {
-				String arg = cmd.substring(4 + i * 5, 8 + i * 5);
+				String arg = cmd.substring(4 + i * (selCom.paramSep ? selCom.paramLen + 1 : selCom.paramLen), 4 + selCom.paramLen + i * (selCom.paramSep ? selCom.paramLen + 1 : selCom.paramLen));
```

Comment:
I'd prefer this big ternary be broken up into a few statements/variables to improve readability